### PR TITLE
Feature/add subcategories controller and view

### PR DESCRIPTION
**Pull request by BogdanLinnik**

### CODE REVIEW

File: app/controllers/admin/tags_controller.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
 class Admin::TagsController < ApplicationController
 
+  def index
+    @tags = Tag.paginate(page: params[:page],
+                         per_page: 2)
```
Comment (illia108) [inline app/controllers/admin/tags_controller.rb]: two tags per page?

File: app/models/category.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class Category < ApplicationRecord
 
   has_many :subcategories
+  has_many :recipes, through: :subcategories
```
Comment (illia108) [inline app/models/category.rb]: nice

File: app/views/admin/categories/show.html.erb [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
-    <% @recipes.each do |recipe| %>
-      <li> <%=link_to recipe.name, admin_recipe_path(recipe)%> </li>
+    <% @subcategories.each do |subcategory| %>
+      <li> <%=link_to subcategory.name, admin_subcategory_path(subcategory)%> </li>
     <% end %>
   </ol>
-<%end%>
+<% end %>
 
 <%= link_to "#{t('button.edit')} #{t('category.name')}",edit_admin_category_path %>
```
Comment (illia108) [inline app/views/admin/categories/show.html.erb]: space before `edit_admin_category_path`

File: app/views/admin/subcategories/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  <div class = "field">
+    <%= f.label :category, "Category"%>
+    <%= f.collection_select :category_id, @categories, :id, :name %>
+  </div>
+
+  <div>
+    <%= f.submit %>
+  </div>
+
+<%end%>
```
Comment (illia108) [inline app/views/admin/subcategories/_form.html.erb]: please surround with spaces

File: app/views/admin/subcategories/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+<%=go_to_subcategories_list%>
```
Comment (illia108) [inline app/views/admin/subcategories/_form.html.erb]: same here

File: app/views/admin/subcategories/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<h1><%= @subcategory.name  %></h1>
+
+<h4><%= t('category.name').capitalize + ": " %> <%= link_to @category.name, admin_category_path(@category) %>
```
Comment (illia108) [inline app/views/admin/subcategories/show.html.erb]: try to put it in one erb closure

File: app/views/admin/subcategories/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+<h5><%= @subcategory.description %></h5>
+
+
+<% if @recipes.empty? %>
+  <h4> <%= t('hasnt').capitalize + " " + t('recipe.names') + " " + t('show.for') + " " + t('show.this') + " " + t('category.name') %> </h4>
+<% else %>
+  <h4> <%= t('subcategory.names').capitalize + " " + t('recipe.names') + ":"%> </h4>
```
Comment (illia108) [inline app/views/admin/subcategories/show.html.erb]: space before %>

File: app/views/admin/subcategories/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  <ol>
+    <% @recipes.each do |recipe| %>
+      <li> <%=link_to recipe.name, admin_recipe_path(recipe)%> </li>
+    <% end %>
+  </ol>
+<% end %>
+
+<%= link_to "#{t('button.edit')} #{t('subcategory.name')}",edit_admin_subcategory_path %>
+
+<%= link_to "#{t('button.delete')} #{t('subcategory.name')}" , admin_subcategory_path(@category), method: :delete %>
```
Comment (illia108) [inline app/views/admin/subcategories/show.html.erb]: confirmation?

File: app/controllers/admin/tags_controller.rb (hunk 1 above)
Comment (BogdanLinnik) [reply app/controllers/admin/tags_controller.rb]: fixed

File: app/views/admin/subcategories/show.html.erb (hunk 6 above)
Comment (BogdanLinnik) [reply app/views/admin/subcategories/show.html.erb]: nope. its simple text and link

File: app/views/admin/subcategories/show.html.erb (hunk 8 above)
Comment (BogdanLinnik) [reply app/views/admin/subcategories/show.html.erb]: Added

File: app/views/admin/subcategories/show.html.erb (hunk 6 above)
Comment (illia108) [reply app/views/admin/subcategories/show.html.erb]: Have you tried?

**Pull request merged**